src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -1035,12 +1042,65 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
                             {}, // leave event
                             stateKey,
                         );
-                        this.scheduleDelayDisconnection(res.delay_id);
+                        this.disconnectDelayId = res.delay_id;
+                    } catch (e) {
+                        // TODO: Retry if rate-limited
+                        logger.error("Failed to prepare delayed disconnection event:", e);
+                    }
+                };
+                await prepareDelayedDisconnection();
+                // Send join event _after_ preparing the delayed disconnection event
+                await this.client.sendStateEvent(
+                    this.room.roomId,
+                    EventType.GroupCallMemberPrefix,
+                    newContent,
+                    stateKey,
+                );
+                // If sending state cancels your own delayed state, prepare another delayed state
+                // TODO: Remove this once MSC4140 is stable & doesn't cancel own delayed state
+                if (this.disconnectDelayId !== undefined) {
+                    try {
+                        await this.client._unstable_updateDelayedEvent(
+                            this.disconnectDelayId,
+                            UpdateDelayedEventAction.Restart,
+                        );
+                    } catch (e) {
+                        // TODO: Retry if rate-limited
+                        if ((<MatrixError>e).errcode === "M_NOT_FOUND") {
```

Comment:
This doesn't work in embedded mode! The widget API apparently doesn't preserve the `errcode` or HTTP response code...

Reply:
Might be worth a comment in-code then

Reply:
Fixed by b070043, which also adds a comment.